src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -130,10 +130,12 @@ private async Task<FormulaValue> TryHandleSet(CallNode node, SymbolContext conte
             return CommonErrors.UnreachableCodeError(node.IRContext);
         }
 
-        public override async ValueTask<FormulaValue> Visit(CallNode node, SymbolContext context)
+        public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorContext context)
         {
             CheckCancel();
-            
+
+            context.StackMarker = context.StackMarker.Inc();
```

Comment:
>ontext.StackMarker = context.StackMarker.Inc(); [](http://example.com/codeflow?start=13&length=47)

Can we avoid this? (see suggestion below) 

The problem is that this is modifying a heap-based object, which means we have to reason about reuse, concurrency, etc.

Reply:
Changed instead to use EvalVisitorContext#IncrementStackDepthCounter everytime it as passed as a parameter in this function. Doesn't mutate state instead creates a new version of EvalVisitorContext with same reference to SymbolContext but with an entirely new StackDepthCounter.